src/trace_comparison.rs

```diff
@@ -110,18 +109,10 @@ pub fn generate_block_comparison(
     let base_block_report = block_report_with_dependencies(&base_traces);
     let native_block_report = block_report_with_dependencies(&native_traces);
 
-    // ToDo: Not taking into account arrays with different length
-    // ToDo: Not taking into account different order in the transactions
-    let post_response: Vec<Value> = match (base_block_report, native_block_report) {
-        (Value::Array(base), Value::Array(native)) => base
-            .into_iter()
-            .zip(native)
-            .map(|(base_tx_trace, native_tx_trace)| {
-                generate_transaction_comparions(base_tx_trace, native_tx_trace)
-            })
-            .collect(),
-        // ToDo: Should we handle this panic accordingly
-        _ => panic!("Expecting arrays "),
+    let post_response = match (base_block_report.as_array(), native_block_report.as_array()) {
+        (Some(base), Some(native)) => compare_traces(base, native),
+        // TODO: Should we handle this panic accordingly
```

Comment:
I think you could just return a Result from `generate_block_comparison` and return Err here. Then in main, where we call this, just match and warn! on the error. That way if you run blockifier-tester and walk away it will still at least continue to cache the base traces.

Reply:
Gotcha that makes sense